app/components/upload_item_shared/index.tsx

```diff
@@ -0,0 +1,313 @@
+                                height: UPLOAD_ITEM_DIMENSIONS.THUMBNAIL_SIZE,
+                                borderRadius: 4,
+                            }}
+                            contentFit='cover'
+                        />
+                    </View>
+                );
+            }
+
+            const imageFileData = {
+                ...fileForCheck,
+                id: file.id,
+                clientId: file.clientId,
+                size: file.size,
+                uri: file.uri,
+                localPath: file.uri,
+                width: file.width,
+                height: file.height,
+                failed: file.failed,
+            } as FileInfo;
+
+            return (
+                <View style={style.imageOnlyThumbnail}>
+                    <ImageFile
+                        file={imageFileData}
+                        forwardRef={forwardRef}
+                        inViewPort={inViewPort}
+                        contentFit='cover'
+                    />
+                </View>
+            );
+        }
+        return (
+            <View style={style.iconContainer}>
+                <FileIcon
+                    iconSize={48}
+                    file={fileForCheck}
+                />
+            </View>
+        );
+    };
+
+    const fileExtension = file.extension?.toUpperCase() || file.name?.split('.').pop()?.toUpperCase() || '';
+    const formattedSize = getFormattedFileSize(file.size || 0);
+
+    const containerStyle = useMemo(() => {
+        let containerStyleType;
+        if (fullWidth && !isImageFile) {
+            containerStyleType = style.fullWidthContainer;
+        } else if (isImageFile) {
+            containerStyleType = style.imageOnlyContainer;
+        } else {
+            containerStyleType = style.fileWithInfoContainer;
+        }
+
+        const baseStyles = [style.previewContainer, containerStyleType];
+
+        if (hasError) {
+            return [
+                ...baseStyles,
+                style.errorBorder,
+            ];
+        }
+
+        return baseStyles;
+    }, [fullWidth, isImageFile, hasError, style.fileWithInfoContainer, style.imageOnlyContainer, style.fullWidthContainer, style.previewContainer, style.errorBorder]);
+
+    return (
+        <View
+            style={containerStyle}
+            testID={testID}
+        >
+            <TouchableWithoutFeedback onPress={onPress}>
+                <Animated.View style={galleryStyles}>
+                    {fileDisplayComponent()}
+                </Animated.View>
+            </TouchableWithoutFeedback>
+
+            {!isImageFile && (
+                <View style={style.fileInfo}>
+                    <Text
+                        style={style.fileName}
+                        numberOfLines={1}
+                        ellipsizeMode='tail'
+                    >
+                        {file.name || 'Unknown file'}
```

Comment:
when do we get `Unknown file` ? and if that the case, shouldn't we have a localization for this ?

Reply:
Yes, nice catch.